src/actions/useActions.ts

```diff
@@ -0,0 +1,44 @@
+import {useClient} from '../hooks/useClient';
+import {useGlobalStore} from '../hooks/useGlobalStore';
+import {CommentData, EntityPointer, SectionData} from '../types';
+
+type UseActionsHookValue = {
+    addCommentToEntity(text: string, entityPointer: EntityPointer): Promise<CommentData>;
+    updateSection(sectionId: string, section: SectionData): Promise<SectionData>;
+}
+
+export const useActions = (): UseActionsHookValue => {
+    const globalStore = useGlobalStore();
+    const client = useClient();
+
+    // TODO: this should be retrieved from globalState
+    // const projectId = globalStore.getCurrentProjectId()
+    const projectId = 'project-1';
+
+    const addCommentToEntity = async (message: string, entityPointer: EntityPointer) => {
+        const commentPayload: Omit<CommentData, 'id'> = {
+            entityType: entityPointer.entityType,
+            entityId: entityPointer.entityId,
+            message,
+            projectId,
+            username: 'mickmister',
+        };
+
+        const comment = await client.addComment(commentPayload);
+        globalStore.addComment(comment);
+
+        return comment;
+    };
+
+    const updateSection = async (sectionId: string, section: SectionData) => {
+        const newSection = await client.updateSection(sectionId, section);
+        globalStore.updateSection(sectionId, newSection);
+
+        return section;
+    };
+
+    return {
+        addCommentToEntity,
+        updateSection,
+    };
```

Comment:
These `actions` functions are called directly from the components. They are the bridge of user interactions to changes to persistent data

Currently there are two `action` functions and two `client` methods that are directly related. There likely won't be a 1-1 relation between them in general as things grow